message_ix/model/MESSAGE/model_core.gms

```diff
@@ -921,23 +923,51 @@ TOTAL_CAPACITY_BOUND_LO(node,inv_tec,year)$( is_bound_total_capacity_lo(node,inv
 ***
 * Equation ACTIVITY_BOUND_UP
 * """"""""""""""""""""""""""
-* This constraint provides lower bounds of a technology activity by mode, summed over all vintages.
-*
+* This constraint provides upper bounds of a technology activity, summed over
+* all vintages. If the mode is 'all', then this constraint is applied across all
+* modes. Otherwise, it is applied to a single, specified mode.
+* 
 *   .. math::
 *      \sum_{y^V \leq y} ACT_{n,t,y^V,y,m,h} \leq bound\_activity\_up_{n,t,m,y,h}
 *
 ***
-ACTIVITY_BOUND_UP(node,tec,year,mode,time)$( map_tec_act(node,tec,year,mode,time)
-        AND is_bound_activity_up(node,tec,year,mode,time) )..
-    SUM(vintage$( map_tec_lifetime(node,tec,vintage,year) ), ACT(node,tec,vintage,year,mode,time) ) =L=
+ACTIVITY_BOUND_UP(node,tec,year,mode,time)$(
+    is_bound_activity_up(node,tec,year,mode,time) AND map_tec_act(node,tec,year,mode,time)
+)..
+    SUM(
+	vintage$( map_tec_lifetime(node,tec,vintage,year) ),
+	ACT(node,tec,vintage,year,mode,time)
+    )
+    =L=
     bound_activity_up(node,tec,year,mode,time)
 %SLACK_ACT_BOUND_UP% + SLACK_ACT_BOUND_UP(node,tec,year,mode,time)
 ;
 
+***
+* Equation ACTIVITY_BOUND_ALL_MODES_UP
+* """"""""""""""""""""""""""""""""""""
+* This constraint provides upper bounds of a technology activity across all modes and vintages.
+*
+*   .. math::
+*      \sum_{y^V \leq y, m} ACT_{n,t,y^V,y,m,h} \leq bound\_activity\_all\_modes\_up_{n,t,y,h}
+*
+***
+ACTIVITY_BOUND_ALL_MODES_UP(node,tec,year,time)$( is_bound_activity_up(node,tec,year,'all',time) )..
+    SUM(
+	(vintage,mode)$( map_tec_lifetime(node,tec,vintage,year) AND map_tec_mode(node,tec,year,mode) ),
+	ACT(node,tec,vintage,year,mode,time)
+    )
+    =L=
+    bound_activity_up(node,tec,year,'all',time)
+%SLACK_ACT_BOUND_UP% + SLACK_ACT_BOUND_UP(node,tec,year,'all',time)
+;
+
 ***
 * Equation ACTIVITY_BOUND_LO
 * """"""""""""""""""""""""""
-* This constraint provides lower bounds of a technology activity by mode summed over all vintages.
+* This constraint provides lower bounds of a technology activity, summed over
+* all vintages. If the mode is 'all', then this constraint is applied across all
```

Comment:
This docstring is still pointing to the `'all'` option which was now moved to an own constraint.